tasks/train-a/sign-in.md

```diff
@@ -42,42 +42,155 @@ User can enter _email_ and _password_ into relative form fields and be granted t
 #### Acceptance Criteria 1: Access control (5)
 
 - User should be authenticated and redirected to the homepage upon entering valid email and password.
-
-- Once signed in, the user should not be able to access the Sign In page again until they log out.
-
-- If an authenticated user tries to access the Sign In page directly via the browser URL, they should be redirected to the [Homepage](./home.md).
-
+- Once signed in, the user should not be able to access the Sign-In page again until they log out.
+- If an authenticated user tries to access the Sign-In page directly via the browser URL, they should be redirected to the [Homepage](./home.md).
 - After successful authentication each following http-requests must comply with [Authorization record](./readme.md#authorization-token)
 
 #### Acceptance Criteria 2: Error message for incorrect password (5)
 
-- Display "_Incorrect email or password_" error message when user enters a valid email but incorrect password.
-
-- User can fill in any symbols in any languages.
-
-- Password has to be not less 8 symbols excluding space around.
+- User can fill any symbols in any languages.
+- Display "_Incorrect email or password_" error message under Email and Password fields when user click "Sign In" button with existing email but incorrect password for that account.
+- Errors appear when user click on "Sign In" button and receive 400 error.
+- Error under field that is being edited disappears when user changes this field content.
 
 #### Acceptance Criteria 3: Error message for email (5)
 
-- Display "_Incorrect email or password_" error message when user enters a non-existent email.
-
-- Display "_Incorrect email_" error message under email field if content of email field does not match _^[\w\d\_]+@[\w\d\_]+\.\w{2,7}$_ regular expression.
+- Display "_Incorrect email or password_" error message under Email and Password fields when user click "Sign In" button with existing email but incorrect password for that account. Error disappears when user changes Email field content.
+- Display "_Incorrect email_" error message under Email field if user changes content of email field after clicking on "Sign In" button and it does not match _^[\w\d\_]+@[\w\d\_]+\.\w{2,7}$_ regular expression.
```

Comment:
here a bit not clear about behaviour - error appears only after user clicks on Sign In button :?
as i understand atm - we shows validation errors if get 400 failed from backend, and we don't show any visible errors if validation fails on frontend side before sending request :?
if so - maybe to rewrite it to:
- Display "_Incorrect email_" error message under Email field if user changes content of email field and clicks after that on "Sign In" button and it does not match _^[\w\d\_]+@[\w\d\_]+\.\w{2,7}$_ regular expression.